github/compliance/issue.py

```diff
@@ -39,47 +43,70 @@ def _issue_labels(
     return frozenset((l.name for l in issue.labels()))
 
 
-def artifact_digest_label(
-    component: cm.Component,
-    artifact: cm.Artifact | str,
-    length: int=18,
-):
+def digest_label(
+    target: gcm.Target,
+    issue_type: str,
+    max_length: int=50,
+) -> str:
     '''
-    calculates and returns a digest for the given component/resource
+    calculates and returns a digest for the given target and issue_type.
 
     this is useful as GitHub labels are limited to 50 characters
     '''
-    if isinstance(artifact, cm.Resource):
-        name = artifact.name
-    else:
-        name = artifact
 
-    label_str = f'{component.name}:{name}'
+    name = target_name(target=target)
```

Comment:
Note:
Previously, only for `cm.Resource` the `.name` attribute was used, in `cm.Source` case (e.g. CheckMarx) the whole object was used instead.

Reply:
I see - that means all cx-issues will likely be re-created after this change, as digest should change, right?
I think that's okay, considering those issues are short-lived anyhow

Reply:
Exactly, we won't be able to find existing CX issues -> they will be re-created.